### ISSUE TITLE
Variable defined inside if block is not marked undefined in else condition

### ISSUE THREAD

**Issue by sirbrillig** [sirbrillig/phpcs-variable-analysis#188]
I haven't actually researched this yet, so this is sort of a placeholder so I don't forget, but I noticed something odd the other day. In this block of code, `$varInfo` is defined inside an `if` block, and then used in the `elseif` condition, which I'm pretty sure is a bug. However, VariableAnalysis didn't report it as an undefined variable.

```php
    if ($referencePtr !== false && $tokens[$referencePtr]['code'] === T_BITWISE_AND) {
      $varInfo = $this->getOrCreateVariableInfo($varName, $currScope);
      $varInfo->isReference = true;
    } elseif ($varInfo->isReference) {
      $this->markVariableRead($varName, $stackPtr, $currScope);
    }
```

**Comment by sirbrillig** [comment]
This is a special case of function scope, because if the variable defined inside an initial `if` block is used inside an `else` (or `elseif`) condition, it would indeed be undefined. However, when used inside another `if` condition, it may be defined.
